src/wallet/rpcwallet.cpp

```diff
@@ -3056,6 +3060,15 @@ UniValue listunspent(const JSONRPCRequest& request)
         entry.pushKV("scriptPubKey", HexStr(scriptPubKey.begin(), scriptPubKey.end()));
         entry.pushKV("amount", ValueFromAmount(out.tx->tx->vout[out.i].nValue));
         entry.pushKV("confirmations", out.nDepth);
+        if (out.nDepth) {
+            LOCK(mempool.cs);
+            const auto it = mempool.mapTx.find(out.tx->GetHash());
```

Comment:
Not sure if this is correct, if `out.nDepth > 0` then `out.tx` should not be in the mempool right? Or should the condition be `out.nDepth == 0`?